etl/steps/data/meadow/climate/2023-12-20/surface_temperature.py

```diff
@@ -0,0 +1,125 @@
+"""Load a snapshot and create a meadow dataset."""
+import gzip
+import zipfile
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from owid.catalog import Table
+from shapely.geometry import mapping
+from structlog import get_logger
+from tqdm import tqdm
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+# Initialize logger.
+log = get_logger()
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("surface_temperature.gz")
+    # Load data from snapshot.
+    with gzip.open(snap.path, "r") as _file:
+        ds = xr.open_dataset(_file)
+
+    #
+    # Process data.
+    #
+    # The latest 3 months in this dataset are made available through ERA5T, which is slightly different to ERA5. In the downloaded file, an extra dimenions ‘expver’ indicates which data is ERA5 (expver = 1) and which is ERA5T (expver = 5).
+    # If a value is missing in the first dataset, it is filled with the value from the second dataset.
+    ERA5_combine = ds.sel(expver=1).combine_first(ds.sel(expver=5))
+
+    # Select the 't2m' variable from the combined dataset and assign it to 'da'.
+    da = ERA5_combine["t2m"]
+
+    # Convert the temperature values from Kelvin to Celsius by subtracting 273.15.
+    da_degc = da - 273.15
+
+    # Read the shapefile to extract country informaiton
```

Comment:
I'd suggest moving the following lines (46-56) to the top of the run function, in the "Load inputs" part.